test/connection-test.ts

```diff
@@ -590,6 +590,26 @@ describe('Connection', function() {
     }
   )
 
+  it('should clean up websocket connection if error after websocket is opened', async function() {
+    await this.api.disconnect();
+    // fail on connection
+    this.api.connection._subscribeToLedger = async () => {
+      throw new Error('error on _subscribeToLedger')
+    }
+    try {
+      await this.api.connect();
```

Comment:
can you throw an "expected to fail" error here, and then check for the correct error below? Something like:

```
try {
  await this.api.connect();
  throw new Error('expected connect() to reject, but it resolved');
} catch (err) {
  assert (err.message === 'error on _subscribeToLedger');
}

this.api.connection._subscribeToLedger = // ...
```

Reply:
Thx, will add.